airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py

```diff
@@ -204,6 +222,136 @@ def test_read_sequential_slices(
                     record_value, state_value, threshold_days
                 ), f"Second incremental sync should produce records older or equal to cursor value from the state. Stream: {stream_name}"
 
+    def test_per_stream_two_sequential_reads(
+        self,
+        inputs: IncrementalConfig,
+        connector_config: SecretDict,
+        configured_catalog_for_incremental: ConfiguredAirbyteCatalog,
+        cursor_paths: dict[str, list[str]],
+        docker_runner: ConnectorRunner,
+    ):
+        threshold_days = getattr(inputs, "threshold_days") or 0
+        stream_mapping = {stream.stream.name: stream for stream in configured_catalog_for_incremental.streams}
+
+        output = docker_runner.call_read(connector_config, configured_catalog_for_incremental)
+
+        records_1 = filter_output(output, type_=Type.RECORD)
+        states_1 = filter_output(output, type_=Type.STATE)
+
+        assert states_1, "Should produce at least one state"
+        assert records_1, "Should produce at least one record"
+
+        # If an outbound state message is not in the per-stream format we should skip the test for the connector being tested
+        if not is_per_stream_state(states_1[-1]):
```

Comment:
An alternative would be turning these tests off with an option in acceptance test YAML. It might be a little annoying to have skipped tests appear for non-per-stream connectors, but this way also gets this to work for free w/o and code changes